### PR TITLE
Init

### PR DESCRIPTION
**Pull request by Anbu-A**
### First _working_ prototype of the TCP-Client

**Form1.cs:** User-input + socket connection
**Form2.cs:** Sending and receiving messages + Disconnecting

### CODE REVIEW

File: Socket_Chat_Client/Socket_Chat_Client/App.config [hunk 1]
```diff
@@ -0,0 +1,6 @@
+﻿<?xml version="1.0" encoding="utf-8" ?>
+<configuration>
+    <startup> 
+        <supportedRuntime version="v4.0" sku=".NETFramework,Version=v4.7.2" />
+    </startup>
+</configuration>
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/App.config:6]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client/Form1.resx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+</root>
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form1.resx:120]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client/Form2.Designer.cs [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+            this.PerformLayout();
+
+        }
+
+        #endregion
+
+        private System.Windows.Forms.TextBox textBox1;
+        private System.Windows.Forms.TextBox textBox2;
+    }
+}
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form2.Designer.cs:74]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client/Form2.resx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+</root>
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form2.resx:120]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client/Properties/Resources.resx [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+  <resheader name="version">
+    <value>2.0</value>
+  </resheader>
+  <resheader name="reader">
+    <value>System.Resources.ResXResourceReader, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+  <resheader name="writer">
+    <value>System.Resources.ResXResourceWriter, System.Windows.Forms, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089</value>
+  </resheader>
+</root>
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Properties/Resources.resx:117]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client/Properties/app.manifest [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+          name="Microsoft.Windows.Common-Controls"
+          version="6.0.0.0"
+          processorArchitecture="*"
+          publicKeyToken="6595b64144ccf1df"
+          language="*"
+        />
+    </dependentAssembly>
+  </dependency>
+  -->
+</assembly>
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Properties/app.manifest:70]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client/Socket_Chat_Client.csproj [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+      <Install>true</Install>
+    </BootstrapperPackage>
+    <BootstrapperPackage Include="Microsoft.Net.Framework.3.5.SP1">
+      <Visible>False</Visible>
+      <ProductName>.NET Framework 3.5 SP1</ProductName>
+      <Install>false</Install>
+    </BootstrapperPackage>
+  </ItemGroup>
+  <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
+</Project>
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Socket_Chat_Client.csproj:140]: newline at the end of a file

File: Socket_Chat_Client/Socket_Chat_Client.sln [hunk 8]
```diff
@@ -0,0 +1,25 @@
+﻿
+Microsoft Visual Studio Solution File, Format Version 12.00
+# Visual Studio Version 16
+VisualStudioVersion = 16.0.30523.141
+MinimumVisualStudioVersion = 10.0.40219.1
+Project("{FAE04EC0-301F-11D3-BF4B-00C04F79EFBC}") = "Socket_Chat_Client", "Socket_Chat_Client\Socket_Chat_Client.csproj", "{5059544E-DEFE-4F8E-97D7-0C4E8C667A03}"
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client.sln:6]: I do not know how this works but is it possible to give the project a non cryptic name?

**Comment by HokageM** [comment]
Give Form1 and Form2 more precise names

File: Socket_Chat_Client/Socket_Chat_Client/Form1.Designer.cs [hunk 9]
```diff
@@ -0,0 +1,146 @@
+﻿namespace Socket_Chat_Client
+{
+    partial class Form1
+    {
+        /// <summary>
+        /// Erforderliche Designervariable.
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form1.Designer.cs:6]: Deutsch -> English in all comments

File: Socket_Chat_Client/Socket_Chat_Client/Form1.cs [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+// Extras for this Project
+using System.Net;
+using System.Net.Sockets;
+
+namespace Socket_Chat_Client
+{
+    public partial class Form1 : Form
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form1.cs:17]: better name

File: Socket_Chat_Client/Socket_Chat_Client/Form1.cs [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+    {
+        String ip_address;
+        int port;
+        bool error_on_connection;
+        IPEndPoint ipe;
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form1.cs:22]: ipe -> ipend_point

non cryptic names are better

e.g.:

What ist "sr"?
->ServerResponse

File: Socket_Chat_Client/Socket_Chat_Client/Form1.cs [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+        Form form = null;
+        public static Form1 instance;
+
+        public Form1()
+        {
+            InitializeComponent();
+            instance = this;
+        }
+
+        private void button1_Click(object sender, EventArgs e)
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form1.cs:34]: button1 -> better name

File: Socket_Chat_Client/Socket_Chat_Client/Form2.cs [hunk 13]
```diff
@@ -0,0 +1,199 @@
...
+
+// Extras for this Project
+using System.Net;
+using System.Net.Sockets;
+using System.Threading;
+
+
+namespace Socket_Chat_Client
+{
+    public partial class Form2 : Form
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form2.cs:19]: from2 better name

File: Socket_Chat_Client/Socket_Chat_Client/Form2.cs [hunk 14]
```diff
@@ -0,0 +1,199 @@
...
+    {
+        String user_name;
+        Socket socket;
+        String msg = "";
+        Thread recv_thread;
+        Form form = null;
+        const String DC_COMMAND = "/dc";
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Form2.cs:26]: also small case to unify the namings

File: Socket_Chat_Client/Socket_Chat_Client/Properties/Resources.Designer.cs [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+//     der Code neu generiert wird.
+// </auto-generated>
+//------------------------------------------------------------------------------
+
+namespace Socket_Chat_Client.Properties
+{
+
+
+    /// <summary>
+    ///   Eine stark typisierte Ressourcenklasse zum Suchen von lokalisierten Zeichenfolgen usw.
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Properties/Resources.Designer.cs:16]: deutsch -> english

File: Socket_Chat_Client/Socket_Chat_Client.sln (hunk 8 above)
Comment (Anbu-A) [reply Socket_Chat_Client/Socket_Chat_Client.sln:6]: I also do not know this but i think we will never have a look at this file again anyway.

File: Socket_Chat_Client/Socket_Chat_Client/Program.cs [hunk 16]
```diff
@@ -0,0 +1,22 @@
...
+using System.Linq;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+namespace Socket_Chat_Client
+{
+    static class Program
+    {
+        /// <summary>
+        /// Der Haupteinstiegspunkt für die Anwendung.
```
Comment (HokageM) [inline Socket_Chat_Client/Socket_Chat_Client/Program.cs:12]: deutsch -> english

**Comment by HokageM** [comment]
I think after this it can be merged :)

**Pull request merged**